### PR TITLE
fixed bug in reorder request queue and move request response

### PR DESCRIPTION
**Pull request by wwelling**

### CODE REVIEW

**Comment by marcjohnson-kint** [comment]
Expedited approval to try to address stability issue in hosted environment

**Comment by mtraneis** [comment]
Can you describe the nature of the bug and what changes are required, please?  Is this for CIRC-316? Hard to follow without much context

**Comment by mtraneis** [comment]
Just saw Marc's comments about expediting approval...

**Comment by marcjohnson-kint** [comment]
> Just saw Marc's comments about expediting approval...

@mtraneis Please feel free to provide further feedback, changes can be made in subsequent pull requests
